go-controller/pkg/ovn/common.go

```diff
@@ -199,16 +199,20 @@ func (oc *Controller) deletePortGroup(hashName string) {
 	}
 }
 
-func (oc *Controller) getIPFromOvnAnnotation(ovnAnnotation string) string {
-	if ovnAnnotation == "" {
+func (oc *Controller) getMacFromOvnAnnotation(ovnAnnotation string) string {
+	ovnAnnotationMap, err := oc.getAnnotationMap(ovnAnnotation)
+	if err != nil {
+		logrus.Errorf("Error retrieving ovn annotation %s from pod object: %s", ovnAnnotation, err.Error())
 		return ""
 	}
 
-	var ovnAnnotationMap map[string]string
-	err := json.Unmarshal([]byte(ovnAnnotation), &ovnAnnotationMap)
+	return ovnAnnotationMap["mac_address"]
+}
+
+func (oc *Controller) getIPFromOvnAnnotation(ovnAnnotation string) string {
```

Comment:
@alexanderConstantinescu let's return an error from this function if the IP doesn't pass `net.ParseCIDR()`